packages/dd-trace/test/plugins/util/inferred_proxy.spec.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const getPort = require('get-port')
+const { expect } = require('chai')
+const axios = require('axios')
+const agent = require('../../../test/plugins/agent')
+
+describe('Inferred Proxy Spans', function () {
+  this.timeout(60000)
+  let port, appListener, controller
+  let http
+
+  before(async () => {
+    process.env.DD_SERVICE = 'aws-server'
+    const tracer = require('../../../../dd-trace').init()
+
+    port = await getPort()
+
+    await agent.load('http').then(() => {
+      http = require('http')
+
+      const server = new http.Server(async (req, res) => {
+        controller && await controller(req, res)
+        res.writeHead(200)
+        res.end(JSON.stringify({ message: 'OK' }))
+      })
+
+      appListener = server.listen(port, 'localhost')
+    })
+  })
+
+  after(() => {
+    appListener && appListener.close()
+    return agent.close({ ritmReset: false })
+  })
+
+  const inferredHeaders = {
+    'x-dd-proxy-name': 'aws-apigateway',
+    'x-dd-proxy-request-time': '123456.45',
+    'x-dd-proxy-path': '/test',
+    'x-dd-proxy-httpmethod': 'GET',
+    'x-dd-proxy-domain-name': 'example.com',
+    'x-dd-proxy-stage': 'dev'
+  }
+
+  it('should create a parent span and a child span', async () => {
```

Comment:
TODO: Add a test for 500 status codes (or any other status code other than 200)

Reply:
this was added